src/main/java/sally/Parser.java

```diff
@@ -0,0 +1,44 @@
+package sally;
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input and returns the corresponding Command object.
+     *
+     * @param input The user's input command.
+     * @return The Command object based on the input.
+     * @throws SallyException If the input command is invalid.
+     */
+    public static Command parse(String input) throws SallyException {
+        // if no space then exception handling
+        if (input.startsWith("list")) {
+            return new ListCommand();
+        } else if (!input.contains(" ")) {
+            throw new SallyException("Sorry, I do not understand the command :-(");
+        } else {
+            String[] parts = input.split(" ", 2);
```

Comment:
Perhaps could be more explanatory with the use of 2 by giving it a variable name?